aiida/repository/backend/abstract.py

```diff
@@ -81,7 +89,7 @@ def has_object(self, key: str) -> bool:
         """
 
     @contextlib.contextmanager
-    def open(self, key: str) -> io.BufferedIOBase:
+    def open(self, key: str) -> typing.Iterator[typing.BinaryIO]:  # type: ignore[return]
```

Comment:
hmmm, these abstract methods, that you also need to call `super()` on, mypy is not a fan 😬